osquery/tables/other/augeas.cpp

```diff
@@ -0,0 +1,91 @@
+/*
+ *  Copyright (c) 2014-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the BSD-style license found in the
+ *  LICENSE file in the root directory of this source tree. An additional grant
+ *  of patent rights can be found in the PATENTS file in the same directory.
+ *
+ */
+#include <osquery/logger.h>
+#include <osquery/tables.h>
+
+#include <augeas.h>
+#include <boost/algorithm/string/join.hpp>
+
+namespace osquery {
+namespace tables {
+void matchAgueasPattern(QueryData &results,
+                        const augeas *aug,
+                        const std::string &pattern) {
+  char **matches = NULL;
+
+  int len = aug_match(aug, pattern.c_str(), &matches);
+
+  // Handle errors
+  if (!matches)
+    return;
+  else if (len < 0) {
+    const char *error_message = aug_error_message(const_cast<augeas *>(aug));
+    const char *error_message_details =
+        aug_error_details(const_cast<augeas *>(aug));
+    VLOG(1) << "An error has occurred while trying to query augeas: "
+            << error_message;
+
+    if (error_message_details)
+      VLOG(1) << "Details: " << error_message_details;
+
+    return;
+  }
+
+  for (int i = 0; i < len; i++) {
+    Row r;
+    const char *path = matches[i];
+    const char *value = nullptr;
+    aug_get(aug, path, &value);
+    r["path"] = path;
+    if (value)
+      r["value"] = value;
+    results.push_back(r);
+
+    // aug_match() allocates the matches and expects the
+    // caller to free each match
+    free((void *)path);
+  }
+  // aug_match() allocates the matches array and expects the
+  // caller to free it
+  free(matches);
+}
+
+QueryData loadAugeas(QueryContext &context) {
+  QueryData results;
+
+  augeas *aug = aug_init(nullptr, nullptr, AUG_NO_ERR_CLOSE);
+
+  // Handle errors
+  if (!aug) {
+    VLOG(1) << "An error has occurred while trying to initialize augeas";
+    return results;
+  } else if (aug_error(aug) != AUG_NOERROR) {
+    VLOG(1) << "An error has occurred while trying to initialize augeas:"
+            << aug_error_message(aug);
+    aug_close(aug);
+    return results;
+  }
+
+  if (context.hasConstraint("path", EQUALS)) {
+    std::string pattern;
+
+    auto path_constraints = context.constraints["path"].getAll(EQUALS);
+    pattern = boost::algorithm::join(path_constraints, "|");
+
+    matchAgueasPattern(results, aug, pattern);
+  } else
+    matchAgueasPattern(results, aug, "/files//*");
+
+  aug_close(aug);
+
+  return results;
+}
+}
+}
```

Comment:
All of these files should have newlines, perhaps configure you IDE to insert newlines for new C++ files created. :)